### ISSUE TITLE
Investigate API and loaded cluster performance with new Feed v1.1

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#10829]
With the introduction of many new rules, decoders and cdb_list in https://github.com/wazuh/wazuh/pull/10676 we need to evaluate the impact of these changes in the Wazuh API and a loaded Wazuh Cluster environment.

Tasks:
- [ ] Compare the performance of API ruleset calls between 4.2 and 4.3 (rules, decoders, cdb_lists)
- [ ] Estimate the impact of modifying and synchronizing the new huge cdb_list file in a big cluster environment (~30 workers)
- [x] What happens with modified cdb_list (default) when an upgrade occurs? To test this, add a custom cdb_list in 4.2 called `uncommon-cmd-opened-process` and upgrade to 4.3 (a new cdb_list `uncommon-cmd-opened-process` is added in 4.3).

**Comment by mcarmona99** [comment]
## Issue update


### What happens with modified CDB lists (default) when an upgrade occurs? 

#### TEST 1

CONTEXT: Wazuh manager with version 4.2.4. I have added `ip_reputation` and `uncommon-cmd-opened-process`. I have configured both in the ruleset section of the manager's `ossec.conf`.

```bash
/var/ossec/etc/lists# ll
total 56
drwxrwx--- 1 root  ossec 4096 Nov 15 16:09 ./
drwxrwx--- 1 ossec ossec 4096 Nov 15 16:09 ../
drwxrwx--- 1 ossec ossec 4096 Nov 15 15:53 amazon/
-rw-rw---- 1 ossec ossec  107 Nov 15 15:49 audit-keys
-rw-rw---- 1 ossec ossec 2265 Nov 15 15:53 audit-keys.cdb
-rw-rw---- 1 ossec ossec  107 Nov 15 15:58 ip_reputation
-rw-rw---- 1 ossec ossec 2265 Nov 15 16:09 ip_reputation.cdb
-rw-rw---- 1 ossec ossec  892 Nov 15 15:49 security-eventchannel
-rw-rw---- 1 ossec ossec 6461 Nov 15 15:53 security-eventchannel.cdb
-rw-rw---- 1 ossec ossec  107 Nov 15 15:59 uncommon-cmd-opened-process
-rw-rw---- 1 ossec ossec 2265 Nov 15 16:09 uncommon-cmd-opened-process.cdb
```

The content of `ip_reputation` and `uncommon-cmd-opened-process` is the following (they are copies of  `audit-keys`):

```bash
# cat /var/ossec/etc/lists/uncommon-cmd-opened-process
audit-wazuh-w:write
audit-wazuh-r:read
audit-wazuh-a:attribute
audit-wazuh-x:execute
audit-wazuh-c:command
```

After upgrading to 4.3.0, the CDB lists `ip_reputation` and `uncommon-cdm-opened-process` have not being overwritten. The user and group permissions have changed from **ossec** to **wazuh** properly.

```bash
# ll /var/ossec/etc/lists/
total 60
drwxrwx--- 1 root  wazuh 4096 Nov 15 16:13 ./
drwxrwx--- 1 wazuh wazuh 4096 Nov 15 16:25 ../
drwxrwx--- 1 wazuh wazuh 4096 Nov 15 16:19 amazon/
-rw-rw---- 1 wazuh wazuh  107 Nov 15 15:49 audit-keys
-rw-rw---- 1 wazuh wazuh 2265 Nov 15 16:12 audit-keys.cdb
-rw-rw---- 1 wazuh wazuh  107 Nov 15 16:12 ip_reputation
-rw-rw---- 1 wazuh wazuh 2265 Nov 15 16:13 ip_reputation.cdb
-rw-rw---- 1 wazuh wazuh  892 Nov 15 15:49 security-eventchannel
-rw-rw---- 1 wazuh wazuh 6461 Nov 15 16:12 security-eventchannel.cdb
-rw-rw---- 1 wazuh wazuh  107 Nov 15 16:12 uncommon-cmd-opened-process
-rw-rw---- 1 wazuh wazuh 2265 Nov 15 16:13 uncommon-cmd-opened-process.cdb
```

```bash
# cat /var/ossec/etc/lists/uncommon-cmd-opened-process
audit-wazuh-w:write
audit-wazuh-r:read
audit-wazuh-a:attribute
audit-wazuh-x:execute
audit-wazuh-c:command
```

#### TEST 2

CONTEXT: Wazuh manager with version 4.2.4. I have added `ip_reputation`. I have configured it in the ruleset section of the manager's `ossec.conf`.

```bash
/var/ossec/etc/lists# ll
total 56
drwxrwx--- 1 root  ossec 4096 Nov 15 16:09 ./
drwxrwx--- 1 ossec ossec 4096 Nov 15 16:09 ../
drwxrwx--- 1 ossec ossec 4096 Nov 15 15:53 amazon/
-rw-rw---- 1 ossec ossec  107 Nov 15 15:49 audit-keys
-rw-rw---- 1 ossec ossec 2265 Nov 15 15:53 audit-keys.cdb
-rw-rw---- 1 ossec ossec  107 Nov 15 15:58 ip_reputation
-rw-rw---- 1 ossec ossec 2265 Nov 15 16:09 ip_reputation.cdb
-rw-rw---- 1 ossec ossec  892 Nov 15 15:49 security-eventchannel
-rw-rw---- 1 ossec ossec 6461 Nov 15 15:53 security-eventchannel.cdb
```

The content of `ip_reputation` is the following (a copy of  `audit-keys`):

```bash
# cat /var/ossec/etc/lists/ip_reputation
audit-wazuh-w:write
audit-wazuh-r:read
audit-wazuh-a:attribute
audit-wazuh-x:execute
audit-wazuh-c:command
```

After upgrading to 4.3.0, the CDB lists `ip_reputation` has not being overwritten. The user and group permissions have changed from **ossec** to **wazuh** properly. Moreover, the `uncommon-cmd-opened-process` CDB list has been added correctly (a new default one added in 4.3.0).

```bash
# ll /var/ossec/etc/lists/
total 56
drwxrwx--- 1 root  wazuh 4096 Nov 15 16:58 ./
drwxrwx--- 1 wazuh wazuh 4096 Nov 15 16:58 ../
drwxrwx--- 1 wazuh wazuh 4096 Nov 15 16:58 amazon/
-rw-rw---- 1 wazuh wazuh  107 Nov 15 15:49 audit-keys
-rw-rw---- 1 wazuh wazuh 2265 Nov 15 16:50 audit-keys.cdb
-rw-rw---- 1 wazuh wazuh  107 Nov 15 16:50 ip_reputation
-rw-rw---- 1 wazuh wazuh 2265 Nov 15 16:51 ip_reputation.cdb
-rw-rw---- 1 wazuh wazuh  892 Nov 15 15:49 security-eventchannel
-rw-rw---- 1 wazuh wazuh 6461 Nov 15 16:50 security-eventchannel.cdb
-rw-rw---- 1 wazuh wazuh   68 Nov 15 16:58 uncommon-cmd-opened-process
```

**Note that this new CDB list is not in the ruleset section of the manager's `ossec.conf` by default (that's why the `.cdb` file is not created after upgrading)**

```bash
# cat /var/ossec/etc/lists/uncommon-cmd-opened-process 
SystemPropertiesAdvanced.EXE:
SystemPropertiesAdvanced.exe:
```

**Comment by mcarmona99** [comment]
This issue is going to be closed as the actual development related to **Feed v1.1** is not going to be released in 4.3.
